ably/realtime_channel.go

```diff
@@ -451,29 +320,27 @@ func (c *RealtimeChannel) notify(msg *proto.ProtocolMessage) {
 	case proto.ActionAttached:
 		c.state.syncSet(StateChanAttached, nil)
 		c.queue.Flush()
+		c.Presence.mu.Lock()
+		c.Presence.syncStart()
+		if msg.Flags&1 == 0 {
+			c.Presence.syncEnd()
```

Comment:
Rather than starting then immediately stopping the sync, why not just only start it if `msg.Flags & 1 == 1`?